eth_xmc4xxx fails to build

**Describe the bug**

In weekly CI, the eth_xmc4xxx ethernet driver fails to build for 
* xmc45_relax_kit
* xmc47_relax_kit

In the following samples and tests
* samples/net/capture/sample.net.capture 
* samples/net/dhcpv4_client/sample.net.dhcpv4_client 
* samples/net/dns_resolve/sample.net.dns_resolve.llmnr 
* samples/net/dns_resolve/sample.net.dns_resolve.mdns 
* samples/net/promiscuous_mode/sample.net.promiscuous_mode 
* samples/net/sockets/big_http_download/sample.net.sockets.big_http_download 
* samples/net/sockets/big_http_download/sample.net.sockets.big_http_download.ci 
* samples/net/sockets/dumb_http_server/sample.net.sockets.dumb_http_server 
* samples/net/sockets/dumb_http_server_mt/sample.net.sockets.dumb_http_server_mt 
* samples/net/sockets/echo/sample.net.sockets.echo 
* samples/net/sockets/echo_service/sample.net.sockets.service.echo 
* samples/net/sockets/http_client/sample.net.sockets.http_client 
* samples/net/sockets/http_get/sample.net.sockets.http_get 
* samples/net/sockets/net_mgmt/sample.net.sockets.net_mgmt.kernelmode 
* samples/net/sockets/net_mgmt/sample.net.sockets.net_mgmt.usermode 
* samples/net/sockets/socketpair/sample.net.sockets.socketpair 
* samples/net/sockets/websocket_client/sample.net.sockets.websocket_client 
* samples/net/stats/sample.net.stats 
* samples/net/syslog_net/sample.net.syslog.ipv4_only 
* samples/net/syslog_net/sample.net.syslog.ipv6_only 
* samples/net/syslog_net/sample.net.syslog.no_autostart 
* samples/net/syslog_net/sample.net.syslog.runtime_srv_addr 
* samples/net/syslog_net/sample.net.syslog.with_timefuncs 
* samples/net/telnet/sample.net.telnet 
* samples/net/virtual/sample.net.virtual 
* samples/net/vlan/sample.net.vlan 
* samples/posix/gettimeofday/sample.posix.gettimeofday 
* tests/drivers/build_all/ethernet/net.ethernet.build.default 
* tests/net/lib/mqtt_publisher/net.mqtt 
* tests/net/lib/mqtt_publisher/net.mqtt.tls 
* tests/net/lib/mqtt_pubsub/net.mqtt.pubsub 
* tests/net/lib/mqtt_pubsub/net.mqtt.pubsub.preempt 
* tests/net/lib/mqtt_subscriber/net.mqtt.subscriber 
* tests/net/lib/mqtt_subscriber/net.mqtt.subscriber.preempt 
* tests/subsys/mgmt/mcumgr/os_mgmt_info/mgmt.mcumgr.os.info.net 

**To Reproduce**
Steps to reproduce the behavior:
1. mkdir build; cd build
2. cmake -GNinja -DBOARD=xmc45_relax_kit ../samples/net/capture/
3. ninja
4. See error

**Expected behavior**
No build failures

**Impact**
weekly CI failure

**Logs and console output**
```
drivers/ethernet/eth_xmc4xxx.c: In function 'eth_xmc4xxx_get_phy':
drivers/ethernet/eth_xmc4xxx.c:642:16: error: 'config' undeclared (first use in this function); did you mean 'mpu_config'?
  642 |         return config->phy_dev;
      |                ^~~~~~
      |                mpu_config
drivers/ethernet/eth_xmc4xxx.c:642:16: note: each undeclared identifier is reported only once for each function it appears in
drivers/ethernet/eth_xmc4xxx.c:640:42: warning: unused variable 'dev_cfg' [-Wunused-variable]
  640 |         const struct eth_xmc4xxx_config *dev_cfg = dev->config;
      |                                          ^~~~~~~
drivers/ethernet/eth_xmc4xxx.c:643:1: warning: control reaches end of non-void function [-Wreturn-type]
  643 | }
      | ^
[207/235] Building C object zephyr/subsys/net/ip/CMakeFiles/subsys__net__ip.dir/tcp.c.obj
ninja: build stopped: subcommand failed.
```

**Environment (please complete the following information):**

 - OS: Linux, Ubuntu 22.04, CI
 - Toolchain: Zephyr SDK 0.16.8
 - Commit SHA or Version used: Main as of now 988e4cf77094fcabaf9a10ad94102d1e4522d0f6

**Additional context**
Introduced by 27aa4bc988d2ebd54a49f4df51ccbc653ebe2d73
#76062

--- a/drivers/ethernet/eth_xmc4xxx.c
+++ b/drivers/ethernet/eth_xmc4xxx.c
@@ -639,7 +639,7 @@ static const struct device *eth_xmc4xxx_get_phy(const struct device *dev)
 {
 	const struct eth_xmc4xxx_config *dev_cfg = dev->config;
 
-	return config->phy_dev;
+	return dev_cfg->phy_dev;
 }
 
 static void eth_xmc4xxx_iface_init(struct net_if *iface)

